wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextStyles.kt

```diff
@@ -0,0 +1,435 @@
+        set(url) = run {
+            action = ClickEvent.Action.OPEN_URL
+            value = url
+        }
+    /** Open's a file. */
+    public var openFile: String = ""
+        set(path) = run {
+            action = ClickEvent.Action.OPEN_FILE
+            value = path
+        }
+    /** Runs a command. */
+    public var runCommand: String = ""
+        set(command) = run {
+            action = ClickEvent.Action.RUN_COMMAND
+            value = command
+        }
+    /** Suggests a command. */
+    public var suggestCommand: String = ""
+        set(suggestedCommand) = run {
+            action = ClickEvent.Action.SUGGEST_COMMAND
+            value = suggestedCommand
+        }
+    /** Change's page. */
+    public var changePage: Int = 0
+        set(page) = run {
+            action = ClickEvent.Action.CHANGE_PAGE
+            value = page.toString()
+        }
+    /** Copies text to the clipboard. */
+    public var copyToClipboard: String = ""
+        set(toCopy) = run {
+            action = ClickEvent.Action.COPY_TO_CLIPBOARD
+            value = toCopy
+        }
+
+    /**
+     * Creates a click event based on the [action] and [value] provided.
+     *
+     * @author NoComment1105
+     */
+    public fun create(): ClickEvent {
+        return ClickEvent(
+            action ?: ClickEvent.Action.SUGGEST_COMMAND,
+            value
+        )
+    }
+}
+
+/**
+ * A mutable style object that transforms to Minecraft [styles][Style].
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+public class MutableStyle {
+    /** A [Color] to apply to the text. */
+    public var color: Color? = null
+    /** Whether to obfuscate the text or not. */
+    public var obfuscated: Boolean = false
+    /** Whether to format the text in bold or not. */
+    public var bold: Boolean = false
+    /** Whether to format the text in italics or not. */
+    public var italic: Boolean = false
+    /** Whether to format the text with a strikethrough or not. */
+    public var strikethrough: Boolean = false
+    /** Whether to format the text with an underline or not. */
+    public var underline: Boolean = false
+    /** A [HoverEvent] to apply to the text. */
+    public var hoverEvent: HoverEvent? = null
+    /** A [ClickEvent] to apply to the text. */
+    public var clickEvent: ClickEvent? = null
+    /**
+     * An insertion inserted when a piece of text clicked while shift key is down in the chat HUD to apply to the text.
+     */
+    public var insertion: String? = null
+    /** An [Identifier] for the Minecraft font that would like to be used. */
+    public var font: Identifier? = null
+
+    /**
+     * Converts 3 separate RGB values to a [Color].
+     *
+     * @param red The red channel of the color
+     * @param green The green channel of the color
+     * @param blue The blue channel of the color
+     *
+     * @author NoComment1105
+     */
+    @Suppress("MagicNumber")
+    public fun color(red: Int, green: Int, blue: Int) {
```

Comment:
You have no `Float` shorthand in `MutableStyle` like you do for the color constructors.